beater/server.go

```diff
@@ -68,34 +54,35 @@ func (s *Server) create(successCallback func([]beat.Event), host string) *http.S
 	}
 
 	return &http.Server{
-		Addr:           host,
+		Addr:           config.Host,
 		Handler:        mux,
-		ReadTimeout:    s.config.ReadTimeout,
-		WriteTimeout:   s.config.WriteTimeout,
-		MaxHeaderBytes: s.config.MaxHeaderBytes,
+		ReadTimeout:    config.ReadTimeout,
+		WriteTimeout:   config.WriteTimeout,
+		MaxHeaderBytes: config.MaxHeaderBytes,
 	}
 }
 
-func (s *Server) Start(successCallback func([]beat.Event), host string) {
-	s.http = s.create(successCallback, host)
-	if s.config.SSL.IsEnabled() {
-		go s.http.ListenAndServeTLS(s.config.SSL.Cert, s.config.SSL.PrivateKey)
+func start(server *http.Server, ssl *SSLConfig) {
+	if ssl.isEnabled() {
+		go server.ListenAndServeTLS(ssl.Cert, ssl.PrivateKey)
 	} else {
-		go s.http.ListenAndServe()
+		go server.ListenAndServe()
 	}
 }
 
-func (s *Server) Stop() error {
+func stop(server *http.Server) error {
 	c := context.TODO()
-	err := s.http.Shutdown(c)
+	err := server.Shutdown(c)
 	return err
 }
 
-func (s *Server) createHandler(p processor.Processor, successCallback func([]beat.Event)) func(w http.ResponseWriter, r *http.Request) {
+type handler func(w http.ResponseWriter, r *http.Request)
+
+func createHandler(p processor.Processor, config Config, publish successCallback) handler {
```

Comment:
On the same page with Silvia that the overall change is more important. At some point we probably need to agree on a style on this otherwise we keep changing it forth and back, but this time is not now.